### PR TITLE
Add device model

### PR DESCRIPTION
**Pull request by djosborne**

### CODE REVIEW

**Comment by djosborne** [comment]
@nahiluhmot

File: app/cappy/models/device.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+module Cappy
+  module Models
+    # This model represents a connected device
+    class Device < ActiveRecord::Base
+      self.table_name = 'devices'
+
```
Comment (nahiluhmot) [inline app/cappy/models/device.rb]: Perhaps we could add another validator that ensures that `last_check_in` >= `created_at`

File: spec/app/cappy/models/device_spec.rb [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+            it 'is not valid' do
+              expect(subject).to_not be_valid
+            end
+          end
+
+          context 'and the last_check_in is not nil' do
+            it 'is valid' do
+              expect(subject).to be_valid
+              subject.save!
```
Comment (nahiluhmot) [inline spec/app/cappy/models/device_spec.rb]: Can you remove this `.save!`? I was using it to test the database_cleaner gem.

**Comment by nahiluhmot** [comment]
@djosborne Two minor comments, looks good otherwise :beers:.

File: app/cappy/models/device.rb (hunk 1 above)
Comment (djosborne) [inline app/cappy/models/device.rb]: I tried by adding this line:
`validates :last_check_in, numericality: { greater_than: :created_at }, unless: "last_check_in.nil?"`

But got the error: ` comparison of Float with DateTime failed`

Can't figure out why - I can push a branch if you want to see

Comment (djosborne) [inline app/cappy/models/device.rb]: Nvm, I think I got it

File: app/cappy/models/device.rb [hunk 3]
```diff
@@ -0,0 +1,23 @@
+module Cappy
+  module Models
+    # This model represents a connected device
+    class Device < ActiveRecord::Base
+      self.table_name = 'devices'
+      valid_device_types = %w(lock outlet gas_valve airbourne_alert)
```
Comment (nahiluhmot) [inline app/cappy/models/device.rb]: `CONSTANTS` should be capitalized

File: spec/app/cappy/models/device_spec.rb [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+    let(:device_id)     { 'DEADBEEF' }
+    let(:name)          { 'Cabinet' }
+    let(:device_type)   { 'lock' }
+    let(:last_check_in) { DateTime.now }
+    let(:created_at)    { DateTime.now - 1 }
+    let(:updated_at)    { DateTime.now - 1 }
+
+    valid_types = %w{lock outlet gas_valve airbourne_alert}
+
+    subject { described_class.new(device_id: device_id,
```
Comment (nahiluhmot) [inline spec/app/cappy/models/device_spec.rb]: Use `do` for multiline blocks.

**Comment by nahiluhmot** [comment]
2 comments, :+1: otherwise.

File: spec/app/cappy/models/device_spec.rb (hunk 4 above)
Comment (tlunter) [inline spec/app/cappy/models/device_spec.rb]: And put anything following the `do` on the next line.

File: app/cappy/models/device.rb [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+
+      validates :device_id,     presence: true
+      validates :name,          presence: true
+      validates :device_type,   presence: true, inclusion: { in: valid_device_types }
+      validates :last_check_in, presence: true
+      validates :created_at,    presence: true
```
Comment (nahiluhmot) [inline app/cappy/models/device.rb]: Can you remove the validations on `created_at` and `updated_at`? `ActiveRecord` handles those, so we shouldn't mess with them.

File: spec/app/cappy/models/device_spec.rb [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+                                  name: name, 
+                                  device_type: device_type,
+                                  last_check_in: last_check_in,
+                                  created_at: created_at,
+                                  updated_at: updated_at ) }
```
Comment (nahiluhmot) [inline spec/app/cappy/models/device_spec.rb]: Same here, don't explicitly set the `created_at` and `updated_at`.

File: db/migrate/20150125183018_create_devices.rb [hunk 7]
```diff
@@ -0,0 +1,14 @@
+class CreateDevices < ActiveRecord::Migration
+  def change
+    create_table :devices do |t|
+      t.string :device_id, null: false
+      t.string :name, null: false
+      t.string :device_type, null: false
+      t.datetime :last_check_in, null: false
+      t.datetime :created_at, null: false
+      t.datetime :updated_at, null: false
```
Comment (nahiluhmot) [inline db/migrate/20150125183018_create_devices.rb]: Don't explicity set a datetime for `created_at` and `updated_at`. Instead, just use `t.timestamps`.

File: spec/app/cappy/models/device_spec.rb (hunk 4 above)
Comment (djosborne) [inline spec/app/cappy/models/device_spec.rb]: Are you talking about the ```subject { described_class```?

**Comment by nahiluhmot** [comment]
@djosborne I realized that I forgot to have rubocop track spec issues as well. Looks like there are a few failures on your branch right now, but the changes look good. :+1: when :green_heart:

**Comment by djosborne** [comment]
@nahiluhmot Good to merge?

**Comment by nahiluhmot** [comment]
@djosborne Yup!

**Pull request merged**